config/passport.js

```diff
@@ -1,186 +1,188 @@
-                if (!user) {
-                    user = new User({
-                        name: profile.displayName,
-                        username: profile.username,
-                        provider: 'twitter',
-                        twitter: profile._json
-                    });
-                    user.save(function(err) {
-                        if (err) console.log(err);
-                        return done(err, user);
-                    });
-                } else {
-                    return done(err, user);
-                }
-            });
+  //Use twitter strategy
+  passport.use(new TwitterStrategy({
+    consumerKey: process.env.TWITTER_CONSUMER_KEY || config.twitter.clientID,
+    consumerSecret: process.env.TWITTER_CONSUMER_SECRET || config.twitter.clientSecret,
+    callbackURL: config.twitter.callbackURL
+  },
+    function (token, tokenSecret, profile, done) {
+      User.findOne({
+        'twitter.id_str': profile.id
+      }, function (err, user) {
+        if (err) {
+          return done(err);
+        }
+        if (!user) {
+          user = new User({
+            name: profile.displayName,
+            username: profile.username,
+            provider: 'twitter',
+            twitter: profile._json
+          });
+          user.save(function (err) {
+            if (err) console.log(err);
+            return done(err, user);
+          });
+        } else {
+          return done(err, user);
         }
-    ));
+      });
+    }
+  ));
 
-    //Use facebook strategy
-    passport.use(new FacebookStrategy({
-            clientID: process.env.FB_CLIENT_ID || config.facebook.clientID,
-            clientSecret: process.env.FB_CLIENT_SECRET || config.facebook.clientSecret,
-            callbackURL: config.facebook.callbackURL
-        },
-        function(accessToken, refreshToken, profile, done) {
-            User.findOne({
-                'facebook.id': profile.id
-            }, function(err, user) {
-                if (err) {
-                    return done(err);
-                }
-                if (!user) {
-                    console.log(profile);
-                    user = new User({
-                        name: profile.displayName,
-                        email: (profile.emails && profile.emails[0].value) || '',
-                        username: profile.username,
-                        provider: 'facebook',
-                        facebook: profile._json
-                    });
-                    user.save(function(err) {
-                        if (err) console.log(err);
-                        user.facebook = null;
-                        return done(err, user);
-                    });
-                } else {
-                    user.facebook = null;
-                    return done(err, user);
-                }
-            });
+  //Use facebook strategy
+  passport.use(new FacebookStrategy({
+    clientID: process.env.FB_CLIENT_ID || config.facebook.clientID,
```

Comment:
If you added environment variables as required above, there'll be no need for `|| config.facebook.clientID`